server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -368,14 +342,11 @@ static void validateTypeName(String type) {
         }
     }
 
-    private synchronized Map<String, DocumentMapper> internalMerge(DocumentMapper mapper,
-                                                                   MergeReason reason) {
+    private synchronized DocumentMapper internalMerge(DocumentMapper mapper, MergeReason reason) {
         boolean hasNested = this.hasNested;
         Map<String, ObjectMapper> fullPathObjectMappers = this.fullPathObjectMappers;
         FieldTypeLookup fieldTypes = this.fieldTypes;
 
-        Map<String, DocumentMapper> results = new LinkedHashMap<>(2);
-
         DocumentMapper newMapper = null;
         if (mapper != null) {
```

Comment:
aside: I think the way this is called by the other internalMapping method now, we should always get a non-null mapper argument. Not entirely sure though, but maybe you can see if you agree and then simplify?

Reply:
You're right, I will simplify this - good catch